src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/HashProviderDispenser.Unix.cs

```diff
@@ -0,0 +1,184 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography;
+using Microsoft.Win32.SafeHandles;
+using Internal.Cryptography;
+
+namespace System.Security.Cryptography
+{
+    internal static partial class HashProviderDispenser
+    {
+        internal static HashProvider CreateHashProvider(string hashAlgorithmId)
```

Comment:
Is this file being carved out to make sharing with Android easier?

I think I'd rather the code was just copied into the new .Android.cs file because Android shouldn't be doing things in terms of Interop.Crypto.

Reply:
Yeah. This PR actually has the beginnings of trying to pull the Android and OpenSSL apart because it is untenable to have Android keep pretending it can be OpenSSL.

HashProviderDispenser.Unix is not actually a new file, Git is just a little confused what is going on.

HashProviderDispenser.OpenSsl and HashProviderDispenser.Android are new. The .Unix one is shared between Android and OpenSsl. I could fully split the two and have no code sharing, but it would be a lot of duplication and make the review bigger.